host/src/raiko.rs

```diff
@@ -113,11 +113,10 @@ impl Raiko {
                 );
 
                 // Make sure the blockhash from the node matches the one from the builder
-                assert_eq!(
-                    Into::<FixedBytes<32>>::into(header.hash().0),
-                    input.block_hash_reference,
-                    "block hash unexpected for block {}",
-                    input.block_number,
+                check_eq(
+                    &Into::<FixedBytes<32>>::into(header.hash().0),
+                    &input.block_hash_reference,
+                    "block hash unexpected",
                 );
```

Comment:
This would be better off as a custom [`RaikoError`](https://github.com/taikoxyz/raiko/blob/8bc4a0f6f2e4339f351894333ccadb0b2582af05/core/src/interfaces.rs#L17) instead of just panic-ing or ignoring the error by just printing. The function already returns a `Result` type.

Reply:
Updated.

Reply:
The newest version doesn't print out the blockhashes when they are different, that printing is very useful for debugging, so would definitely switch to something here that still prints it out.

Maybe we should have a `check_eq` that doesn't return error (current version) and also a `require_eq` which does return an error that can be propagated. `check_eq` could be implemented using `require_eq` by simply not propagating the error but just printing the error out.

Reply:
Unesolving this one because the correct/incorrect blockhash are currently still not logged, and I think the approach I gave above would work nicely to make that work in a general way.

Reply:
Done.